boards/DCM/Src/App/App_SetPeriodicCanSignals.c

```diff
@@ -0,0 +1,65 @@
+#include "App_SetPeriodicCanSignals.h"
+#include "App_DcmWorld.h"
+#include "App_CanRx.h"
+#include "App_CanTx.h"
+
+// Regen allowed when braking or (speed > 5kmh and AIRs closed)
+static bool RegenAllowed(struct DcmCanRxInterface *can_rx)
+{
+    float threshold_speed = 5.0f; // 5kmh
+    bool  air_closed =
+        (App_CanRx_BMS_AIR_STATES_GetSignal_AIR_POSITIVE(can_rx) ==
+         CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE) &&
+        (App_CanRx_BMS_AIR_STATES_GetSignal_AIR_NEGATIVE(can_rx) ==
+         CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE);
+    return App_CanRx_FSM_BRAKE_GetSignal_BRAKE_IS_ACTUATED(can_rx) ||
+           ((App_CanRx_FSM_WHEEL_SPEED_SENSOR_GetSignal_LEFT_WHEEL_SPEED(
+                 can_rx) > threshold_speed) &&
+            (App_CanRx_FSM_WHEEL_SPEED_SENSOR_GetSignal_RIGHT_WHEEL_SPEED(
+                 can_rx) > threshold_speed) &&
+            air_closed);
+}
+
+// TODO: Implement PID controller to maintain DC bus power at 80kW
```

Comment:
Create an issue for this and reference it in this TODO

Reply:
#680 done

Reply:
perfect